hexa/databases/tests/test_utils.py

```diff
@@ -0,0 +1,142 @@
+            "sabrina@bluesquarehub.com",
+            "standardpassword",
+        )
+        FeatureFlag.objects.create(
+            feature=Feature.objects.create(code="workspaces"), user=cls.USER_SABRINA
+        )
+
+        cls.WORKSPACE = Workspace.objects.create_if_has_perm(
+            cls.USER_SABRINA,
+            name="Test Workspace",
+            description="Test workspace",
+            countries=[],
+        )
+
+        cls.WORKSPACE = Workspace.objects.create_if_has_perm(
+            cls.USER_SABRINA,
+            name="Test Workspace",
+            description="Test workspace",
+            countries=[],
+        )
+        setattr(cls.WORKSPACE, "database", cls.DB1)
+        WorkspaceMembership.objects.create(
+            user=cls.USER_SABRINA,
+            workspace=cls.WORKSPACE,
+            role=WorkspaceMembershipRole.VIEWER,
+        )
+
+    @mock.patch("psycopg2.connect")
+    def test_get_database_tables_empty(self, mock_connect):
+        mock_connection = mock_connect.return_value
+        mock_context_object = mock_connection.__enter__.return_value
+        cursor = mock_context_object.cursor.return_value.__enter__.return_value
+        cursor.fetchall.return_value = []
+
+        result = get_database_definition(self.WORKSPACE)
+        self.assertEqual(0, len(result))
+
+    @mock.patch("psycopg2.connect")
+    def test_get_database_tables(self, mock_connect):
+        table_name = "database_tutorial"
+        row_count = 2
+        table = {"name": table_name, "columns": [], "count": row_count}
+
+        mock_connection = mock_connect.return_value
+        mock_context_object = mock_connection.__enter__.return_value
+        cursor = mock_context_object.cursor.return_value.__enter__.return_value
+        cursor.fetchall.return_value = [table]
+        cursor.fetchone.return_value = {"row_count": 2}
+
+        result = get_database_definition(self.WORKSPACE)
+
+        self.assertEqual(1, len(result))
+        self.assertEqual(table_name, result[0]["name"])
+
+    @mock.patch("psycopg2.connect")
+    def test_get_table_definition(self, mock_connect):
+        table_name = "database_tutorial"
+        row_count = 2
+        schema = {"name": "id", "type": "int"}
+        table = {
+            "name": table_name,
+            "columns": [schema],
+            "count": row_count,
+            "workspace": self.WORKSPACE,
+        }
+
+        mock_connection = mock_connect.return_value
+        mock_context_object = mock_connection.__enter__.return_value
+        cursor = mock_context_object.cursor.return_value.__enter__.return_value
+        cursor.fetchall.return_value = [schema]
+        cursor.fetchone.return_value = {"row_count": row_count}
+
+        result = get_table_definition(self.WORKSPACE, table_name)
+
+        self.assertEqual(table, result)
+
+    @mock.patch("psycopg2.connect")
+    def test_get_table_definition_not_found(self, mock_connect):
```

Comment:
@pvanliefland Is there a better way to mock a context manager ?

Reply:
Not that I know of, sorry